controllers/vgmanager.go

```diff
@@ -62,6 +56,9 @@ func (v vgManager) ensureCreated(r *LVMClusterReconciler, ctx context.Context, l
 	// the anonymous mutate function modifies the daemonset object after fetching it.
 	// if the daemonset does not already exist, it creates it, otherwise, it updates it
 	result, err := ctrl.CreateOrUpdate(ctx, r.Client, ds, func() error {
+		if err := ctrl.SetControllerReference(lvmCluster, ds, r.Scheme); err != nil {
```

Comment:
It is already set in the line 87.

Reply:
that explains why it worked before :D ill remove it from 87 and put it here because then the create/update doesnt need to define the call twice